src/libleo/leo/leomotor.c

```diff
@@ -0,0 +1,37 @@
+#include "global.h"
+#include "ultra64/leo.h"
+#include "ultra64/leoappli.h"
+#include "ultra64/leodrive.h"
+
+void leoStart_stop(void) {
+    u32 send_cmd;
+    u8 sense_code;
+    u8 retry_cntr = 8;
+    u32 send_data;
+
+    do {
+        send_data = 0;
+        if ((LEOcur_command->header.control & 1)) {
+            send_cmd = 0x50001;
+        } else if ((LEOcur_command->header.control & 2)) {
+            send_cmd = 0xD0000;
+        } else {
+            if ((LEOcur_command->header.control & 4)) {
+                send_data = 0x10000;
+            }
+            send_cmd = 0x40000;
+        }
+        sense_code = leoSend_asic_cmd_w_nochkDiskChange(send_cmd, send_data);
+        if (sense_code == 0) {
+            LEOcur_command->header.status = 0;
+            return;
+        }
+        if (leoChk_err_retry(sense_code)) {
+            break;
+        }
+    } while (retry_cntr--);
+
+    LEOcur_command->header.sense = sense_code;
+    LEOcur_command->header.status = 2;
+    return;
```

Comment:
Does the explicit return matter? I've seen them matter at low optimization levels but all of libleo is -O2 right?

Reply:
I suspect it's here because it's required to match libleo_d: https://decomp.me/scratch/CpWYb

I personally only matched debug for the functions I decomped when I thought it would help with the -O2 version. I'm not sure how consistent the rest of the code is, and we won't have any mechanism to check this. Should we get rid of this and only worry about -O2?

Reply:
I don't have very strong opinions on this, but one thought is that since we don't need to match the debug library some later cleanup pass could come and remove these with no consequences. So maybe in that view they should just not be included from the start? idk, maybe this could do with more opinions.

Reply:
Yeah I agree it's confusing with no way to check, and I doubt all the rest of the code matches debug anyway. I removed it, it's not hard to figure out we need to add this back if we do want to match debug in the future